Colorize the tags!

michael-hawker opened the pull request:
Has a contrast issue with foreground text though:

![image](https://github.com/user-attachments/assets/ebc39efd-f796-408b-b5a1-c423ff9ea7d9)

@niels9001 think we should have a 'smart' thing that tries to create a contrasting brush for the text based on the background brush or would it be best to let devs config this as part of the tag?

For some reason my pokedex sample extension isn't loading, so that'd be the real test for colors... 😆

<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [ ] **Closes:** #xxx
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

michael-hawker commented:
Built on #228 now, just caches brushes in the UI and colors them nicely now with a simple x:Bind function, like we did for icons

michael-hawker commented:
Have one small fix to commit for icon, then we look like this:
![image](https://github.com/user-attachments/assets/05e4ff16-aa3b-46d7-a45a-3c705fffcb5e)

zadjii-msft commented on src/modules/cmdpal/exts/PokedexExtension/Properties/launchSettings.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "profiles": {
+    "MastodonExtension (Package)": {
```
Op this needs to say 
```suggestion
    "PokedexExtension (Package)": {
```

(and below)

Otherwise folks get real confused

zadjii-msft commented on src/modules/cmdpal/exts/VirtualDesktopExtension/Properties/launchSettings.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "profiles": {
+    "MastodonExtension (Package)": {
```
same issue here about the name

zadjii-msft commented on src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml:
```diff
@@ -46,13 +46,15 @@
                 <TextBlock
                     VerticalAlignment="Center"
                     FontSize="12"
+                    Foreground="{x:Bind help:OptionalColorBrushCacheProvider.Convert(Foreground, x:True), Mode=OneWay}"
```
can we like, pass in `ActualTheme` through the binding to let the thing know which theme to be in?

michael-hawker replied:
Yeah, we could do that. I think based on the other comment we just need to figure out what the patterns we want here are for the various scenarios. Then we can figure out how to adjust this to support it after. Let's follow-up on this as a follow-on to the initially enablement here.

zadjii-msft commented on src/modules/cmdpal/Microsoft.CmdPal.UI/Helpers/OptionalColorToBrushConverter.cs:
```diff
@@ -0,0 +1,36 @@
+
+    public static SolidColorBrush? Convert(OptionalColor color, bool isForeground)
+    {
+        if (!color.HasValue)
```
I almost think we want to pass in both the foreground and background OptionalColors here, cause:

* If there's a foreground but no background, we want the text and border to be the FG color
* if there's no FG or BG, we want the text to be default colored, and the border to be hidden (so it's just some text)
* if there's a background but no foreground, I think it's okay to have the text be "transparent" / punch out
  * You know, as soon as  typed that up, I remember that's incredibly difficult to actually do. 

okay you know what, maybe that's harder than I thought

michael-hawker replied:
@niels9001 want to think about these scenarios a bit? I can sync with you this week on what other types of adjustments we'd want to configure on this and then how we can tackle it from a converter/function/helper point of view (as this ties into all the same issues we've been facing with the discussion on the color ramp helpers in the Toolkit as well).

zadjii-msft reviewed (commented):
thoughts, but nothing definitive. The "punch out" is tricky...

michael-hawker commented:
@zadjii-msft rebased on main and addressed the main issues with the mismatched launch settings (also added missing extensions to the clean-up bat).

Let's address more improvements to the tag coloring as a follow-on.
